pkg/operatorsource/phase_reconciler_strategy.go

```diff
@@ -0,0 +1,59 @@
+package operatorsource
+
+import (
+	"fmt"
+
+	"github.com/operator-framework/operator-marketplace/pkg/apis/marketplace/v1alpha1"
+	"github.com/operator-framework/operator-marketplace/pkg/appregistry"
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+)
+
+// PhaseReconcilerStrategy is the interface that wraps the GetPhaseReconciler method
+//
+// GetPhaseReconciler returns an appropriate phaseReconciler based on the current phase of an OperatorSource object.
+// The following chain shows how an OperatorSource object progresses through a series of transitions
+// from the initial phase to complete reconciled state.
+//
+//  Initial --> Validating --> Downloading --> Configuring --> Succeeded
+//  On error, the object is transitioned into "Failed" phase.
+//
+type PhaseReconcilerStrategy interface {
```

Comment:
I am unable to understand why this is a strategy. This looks more like a factory.